api/src/controllers/news-controller.js

```diff
@@ -40,47 +44,79 @@ exports.protectedPut = async function (args, res, next) {
     queryActions.sendResponse(res, 400, {});
     next();
   }
-  let news = null;
+  let incomingObj = {};
   if (args.swagger.params.news && args.swagger.params.news.value) {
-    news = args.swagger.params.news.value
+    incomingObj = args.swagger.params.news.value
   } else {
     defaultLog.info(`protectedPut - you must provide an object`);
     queryActions.sendResponse(res, 400, {});
     next();
   }
 
+  let news = {
+    $set: {
+      projectName: incomingObj.projectName,
+      description: incomingObj.description,
+      url: incomingObj.url,
+      title: incomingObj.title,
+      _epicProjectId: new ObjectID(incomingObj._epicProjectId),
+      type: incomingObj.type,
+      date: new Date(incomingObj.date)
+    }
+  };
+
+  let obj = null;
   try {
-    await Put.updateById(newsId, news);
+    obj = await Put.updateById(newsId, news);
   } catch (error) {
     defaultLog.info(`protectedPut - error updating news: ${newsId}`);
     defaultLog.debug(error);
-    return queryActions.sendResponse(res, 400, {});
+    console.log(news);
+    console.log(error);
+    return queryActions.sendResponse(res, 400, error);
   }
 
-  queryActions.sendResponse(res, 200, {});
+  queryActions.sendResponse(res, 200, obj);
   next();
 }
 
 exports.protectedPost = async function (args, res, next) {
-  let news = null;
+  let incomingObj = {};
   if (args.swagger.params.news && args.swagger.params.news.value) {
-    news = args.swagger.params.news.value
+    incomingObj = args.swagger.params.news.value
   } else {
     defaultLog.info(`protectedPost - you must provide an id to post`);
     queryActions.sendResponse(res, 400, {});
     next();
   }
-  news['_schemaName'] = 'ActivityLNG';
 
+  const ActivityLNG = mongoose.model('ActivityLNG');
+  let news = new ActivityLNG();
+  news._schemaName = 'ActivityLNG';
+
+  incomingObj._epicProjectId &&
+    ObjectID.isValid(incomingObj._epicProjectId) &&
+    (news._epicProjectId = new ObjectID(incomingObj._epicProjectId));
+
+  incomingObj.description && (news.description = incomingObj.description);
+  incomingObj.projectName && (news.projectName = incomingObj.projectName);
+  incomingObj.type && (news.type = incomingObj.type);
+  incomingObj.type && (news.type = incomingObj.type);
```

Comment:
Looks like a duped line. I'm guessing it should be:
`incomingObj.title && (news.type = incomingObj.title);`

Reply:
3a32d6f8c9e96a7ef201e56a13995a6357f2498e